### PR TITLE
Native Client platform & compiler identifiers

### PR DESCRIPTION
**Pull request by suertreus**

### CODE REVIEW

File: Compilers.md [hunk 1]
```diff
@@ -809,6 +809,12 @@ Compiler version|Solaris Studio|`__SUNPRO_C`
 
 The name of Oracle Solaris Studio has changed over the years (e.g. Sun Studio, Sun Workshop, Forte Developer) but we do not make this distinction in the table above.
 
+## [PNaCl](https://en.wikipedia.org/wiki/Google_Native_Client) ##
```
Comment (a1batross) [inline Compilers.md]: Is PNaCl a compiler? 

Looks like it's a normal compiler based on Clang but adds PNaCl as a target. Shouldn't it be added to Architectures instead?

File: OperatingSystems.md [hunk 2]
```diff
@@ -354,6 +354,12 @@ Identification|`__MSDOS__`
 Identification|`_MSDOS`
 Identification|`__DOS__`
 
+## [Native Client](https://en.wikipedia.org/wiki/Google_Native_Client) ##
+
+Type|Macro
+---|---
+Identification|`__native_client__`
```
Comment (a1batross) [inline OperatingSystems.md:361]: So NaCl compiles to native code? It's probably ok to be in OperatingSystems then.

File: Compilers.md (hunk 1 above)
Comment (suertreus) [reply Compilers.md]: > Shouldn't it be added to Architectures instead?
Done.

These are both hard to categorize, but I think Architectures makes sense 👍.

File: OperatingSystems.md (hunk 2 above)
Comment (suertreus) [reply OperatingSystems.md:361]: I think OperatingSystems is the least bad fit for this.  You can build for `__native_client__` + x86 or `__native_client__` + ARM, so I don't think it's an architecture exactly.

Comment (a1batross) [reply OperatingSystems.md:361]: Yeah, that makes sense. Thank you!

**Pull request merged**